tests/NCronJob.Tests/TestHelper.cs

```diff
@@ -51,11 +55,44 @@ protected virtual void Dispose(bool disposing)
         cancellationTokenSource.Cancel();
         cancellationTokenSource.Dispose();
         cancellationSignaled.TrySetCanceled();
+
+        foreach (IDisposable disposable in disposables)
+        {
+            disposable.Dispose();
+        }
+
         serviceProvider?.Dispose();
-    }
 
-    // TODO: Replace calls to this method in the tests with `ServiceProvider`
-    protected ServiceProvider CreateServiceProvider() => ServiceProvider;
+        var current = TestContext.Current;
+
+        if (current.Warnings is not null)
+        {
+            current.TestOutputHelper!.WriteLine("** Warnings:");
+
+            foreach (string warning in current.Warnings)
+            {
+                current.TestOutputHelper!.WriteLine(warning);
+            }
+        }
+
+        if (current.TestState is not null && current.TestState.Result == TestResult.Failed)
```

Comment:
@linkdotnet Added this as a way to get a better understanding at test failures during CI builds.

As the underlying hardware of the agents we run the tests on are very different, we may trigger race concurrency issues.
When that happens, the test will fail and we dump in the test log output the current test context (Storage.TimedEntries and Events) in order to get a better view of what actually resulted from the test execution.

Reply:
I absolutely like that